### PR TITLE
functor task 1, VNevzorov

### PR DESCRIPTION
**Pull request by sam-mcdonalds**

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -1,48 +1,81 @@
...
- * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
+ * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe ->
+ * HoeGaarden...
  */
-struct BeerOrganizer
-{
+struct BeerOrganizer {
+  // we can always be sure that generated beerBrand will be strictly between
+  // None and Max exclusive
+  BeerOrganizer() : m_beer{BeerBrand::None} {}
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:17]: Prefer in-place initialization and follow rule of zero

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -1,48 +1,81 @@
...
- * @todo Implement unary predicate function that will determine if beer is expensive
+ * @todo Implement unary predicate function that will determine if beer is
+ * expensive
  * @return true in case if beer is expensive, false if not
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
-{
+bool isExpensiveBeer(BeerBrand beer) {
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:38]: _don't forget about const :)_

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -1,48 +1,81 @@
...
+   * Gin + Lime juice = Gimlet;
+   * Gin + Grapefruit juice = Greyhount;
+   * Whiskey + SevenUp = SevenPlusSeven;
+   * Others + Others = Oops;
+   */
+
+  static std::map<std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail>
+      cocktailsRecipes;
+
+  static Cocktail mix(AlcoholDrink ad, NonAlcoholDrink nad) {
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:65]: _use more descriptive names, no abbreviations_

File: functors task (1)/student/include/impl.h [hunk 4]
```diff
@@ -1,48 +1,81 @@
...
+    if (cocktailsRecipes.count({ad, nad}) > 0) {
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:66]: Map is a good alternative for a switch here! 
Nice :)

File: functors task (1)/student/include/impl.h (hunk 2 above)
Comment (sam-mcdonalds) [reply functors task (1)/student/include/impl.h]: Does it is really needed when we passed arguments by value?
I mean, the caller is anyway guaranteed that the original value won't change.

Comment (smay1613) [reply functors task (1)/student/include/impl.h]: Just a style: always mark function params as const, because you can modify it accidentally

**Pull request closed without merging**